### DISCUSSION TITLE
GPU-CPU acceleration

### DISCUSSION THREAD

**Discussion by ddeepwell** [graphia-app/graphia#35] [Q&A] [answered]
Does graphia support hardware acceleration from parallel computing or the use of GPUs?

**Comment by timangus** [comment, marked as answer]
Not in any meaningful way. There is a small amount of GPU acceleration used in the text rendering, but otherwise no. This is a slightly embarrassing state of affairs really; the general plan was to create CPU versions of the heavy algorithms first and *at some point* do GPU versions later, but the need has never really properly surfaced. At least for most users it's generally seen as fast enough, it seems.

**Reply by ddeepwell** [reply]
Thanks for the feedback. I am looking to visualize a large node network (over a million nodes and edges) in an interactive manner. The only way I see this happening is with hardware acceleration. I am aware that a visual this large will be difficult to digest because of the complexity of elements, but in-and-of-itself, a visual of this proportion will be useful for a sense of scale.
I suppose I should consider some form of data aggregation and filtering before attempting something so large on Graphia.

**Reply by timangus** [reply]
Ah well the visualisation itself is hardware accelerated. I thought you were referring to using the GPU to do general purpose computing (GPGPU). 1 million node/edge graphs are within the capabilities of Graphia, assuming you have a reasonably good spec graphics card and enough system memory to store all the graph data, but it will be starting to push the limits of what is possible on current hardware.

**Reply by ddeepwell** [reply]
Ok, so Graphia will naturally use the underlying GPUs in large scale visualizations?
And to confirm, it sounds like there is no cpu parallelizations?

**Reply by timangus** [reply]
Yes, it will use hardware acceleration for graphics.
For most of the heavier algorithms, CPU parallelism *is* used.
There is currently very little GPGPU acceleration employed (which is what I thought you were originally asking).
